refactor abstract, intro and overview to make it tighter (#38)
yaronf/draft-tls-attestation: draft-fossati-tls-attestation.md

normative:
  RFC2119:
  RFC8446: tls13
  I-D.ietf-rats-msg-wrap: cmw
informative:
  I-D.bft-rats-kat: rats-kat
  RFC9334: rats-arch
  I-D.ietf-rats-eat: rats-eat

>   I-D.ietf-rats-ar4si: rats-ar4si

Author: ```suggestion
  I-D.ietf-rats-ar4si: rats-ar4si
  I-D.ietf-teep-architecture: teep-arch
```

--- abstract

Attestation is the process by which an entity produces evidence about itself that another party can use to evaluate the trustworthiness of that entity.

> This document describes a series of protocol extensions to the TLS 1.3 handshake that enables the binding of the TLS authentication key to a remote attestation session.

Reviewer: The binding to an attestation *session* is an artifact of the solution, because what we want to attest is a platform.

More generally, I think the new text is much more difficult to read than the old one (4 sentences that depend on one another in a logical sequence). Maybe the old text should be polished a little bit, but I still prefer it as a starting point.

Author: > The binding to an attestation session is an artifact of the solution, because what we want to attest is a platform.

I don't think of it as an emerging property.  It's a precise goal.

Reviewer 2: I don't find it as difficult, but I think reshuffling a bit would make it nicer to read. For example, putting the current line 120 (about attestation giving you more comprehensive security metrics) immediately after L117.

For the TLS part (lines 118-119 at the moment) maybe it'd be easier to follow, particularly by people not familiar with these technologies, if we start by stating the goal of "attestation for authentication during setup of TLS channel".

Author: In the call we thought that, given all the tweaking that occurred in the last two weeks, this is now resolved.

--- middle

#  Introduction

Attestation is the process by which an entity produces evidence about itself that another party can use to evaluate the trustworthiness of that entity.
This document describes a series of protocol extensions to the TLS 1.3 handshake that enables the binding of the TLS authentication key to a remote attestation session.
As a result, a peer can use "attestation credentials", consisting of compound platform evidence and key attestation, to authenticate itself to its peer during the setup of the TLS channel.
This enables an attester, such as a confidential workload running in a Trusted Execution Environment (TEE), or an IoT device that is trying to authenticate itself to a network access point, to present a more comprehensive set of security metrics to its peer.
This, in turn, allows for the implementation of authorization policies at the relying parties that are based on stronger security signals.

Given the variety of deployed and emerging attestation technologies (e.g., {{TPM1.2}}, {{TPM2.0}}, {{-rats-eat}}) these extensions have been explicitly designed to be agnostic of the attestation formats.

> This is achieved by reusing the generic encapsulation defined in {{-cmw}} for transporting evidence and attestation result payloads in the TLS Certificate message.

Reviewer: ```suggestion
This is achieved by reusing the generic encapsulation defined in {{-cmw}} for transporting evidence and attestation result payloads in the TLS Certificate message.

```

The proposed design supports both background-check and passport topologies, as described in {{Sections 5.2 and 5.1 of -rats-arch}}.
This is detailed in {{evidence-extensions}} and {{attestation-results-extensions}}.
This specification provides both one-way (server-only) and mutual (client and server) authentication using attestation credentials, and allows the attestation topologies at each peer to be independent of each other.

This document does not specify any attestation technology.
Companion documents are expected to define specific attestation mechanisms.

# Conventions and Terminology

The reader is assumed to be familiar with the vocabulary and concepts defined in

> {{Section 4 of -rats-arch}}, and those in {{Section 2 of -rats-kat}}.

Reviewer: We cannot say that the use of KAT is optional and then also assume that people have read it.

Author: KAT is not optional, it's informative.  This is not dissimilar to -rats-arch, which is also an informative ref.

Reviewer 2: This is one of the reasons why I'm starting to think we're overloading our KAT draft. In reality, we probably want a document talking about "architectures" or approaches to key attestation, which defines the problem and the types of solutions you might get (e.g., delegated attestation vs "integrated"?), and another document just to describe an EAT-based format. For this draft we'd only need an info reference to the former.

Author: This is now tracked in #41

--- abstract

The TLS handshake protocol allows authentication of one or both peers using static, long-term credentials.

> In some cases, it is also desirable to ensure that the peer runtime environment is in a secure state.
> Attestation is the process by which an entity produces evidence about itself that another party can use to evaluate the trustworthiness of that entity.

Author: ```suggestion
In some cases, it is also desirable to ensure that the peer runtime environment is in a secure state.
Attestation is the process by which an entity produces evidence about itself that another party can use to appraise whether that entity is found in a secure state.
```

This document describes a series of protocol extensions to the TLS 1.3 handshake that enables the binding of the TLS authentication key to a remote attestation session.

> This enables an attester, such as a confidential workload running in a Trusted Execution Environment (TEE), or an IoT device that is trying to authenticate itself to a network access point, to present a more comprehensive set of security metrics to its peer.

Author: ```suggestion
This enables such an entity, for example, a confidential workload running in a Trusted Execution Environment (TEE), or an IoT device that is trying to authenticate itself to a network access point, to present a more comprehensive set of security metrics to its peer.
```

> Attestation is the process by which an entity produces evidence about itself that another party can use to evaluate the trustworthiness of that entity.
> This document describes a series of protocol extensions to the TLS 1.3 handshake that enables the binding of the TLS authentication key to a remote attestation session.
> As a result, a peer can use "attestation credentials" to authenticate itself to its peer during the setup of the TLS channel.
> This enables an attester, such as a confidential workload running in a Trusted Execution Environment (TEE), or an IoT device that is trying to authenticate itself to a network access point, to present a more comprehensive set of security metrics to its peer.
> This, in turn, allows for the implementation of authorization policies at the relying parties that are based on stronger security signals.
> These extensions have been designed to allow the peers to use any attestation technology, in any remote attestation topology, and mutually.

Author: ```suggestion
Attestation is the process by which an entity produces evidence about itself that another party can use to evaluate the trustworthiness of that entity.
Our goal is to use attestation to authenticate one or both peers of a TLS connection.
This document describes a series of protocol extensions to the TLS 1.3 handshake that enables the binding of the TLS authentication key to a remote attestation session.
This enables an attester, such as a confidential workload running in a Trusted Execution Environment (TEE), or an IoT device that is trying to authenticate itself to a network access point, to present a more comprehensive set of security metrics to its peer.
This, in turn, allows for the implementation of authorization policies at the relying parties that are based on stronger security signals.
These extensions have been designed to allow the peers to use any attestation technology, in any remote attestation topology, and mutually.
```

--- middle

#  Introduction

> Attestation is the process by which an entity produces evidence about itself that another party can use to evaluate the trustworthiness of that entity.

Author: ```suggestion
Attestation {{-rats-arch}} is the process by which an entity produces evidence about itself that another party can use to evaluate the trustworthiness of that entity.
```

This document describes a series of protocol extensions to the TLS 1.3 handshake that enables the binding of the TLS authentication key to a remote attestation session.

> As a result, a peer can use "attestation credentials", consisting of compound platform evidence and key attestation, to authenticate itself to its peer during the setup of the TLS channel.

Reviewer 2: You seem to skip over attestation results, unless you use "key attestation" to mean either evidence or results?

Author: > You seem to skip over attestation results, unless you use "key attestation" to mean either evidence or results?

yes.

> This enables an attester, such as a confidential workload running in a Trusted Execution Environment (TEE), or an IoT device that is trying to authenticate itself to a network access point, to present a more comprehensive set of security metrics to its peer.

Reviewer 2: Since you're reworking this: is this duplication between abstract and intro somewhat expected, using them as two completely unrelated chunks of information?

Author: now that we tweaked it, are they sufficiently different?

--- abstract

The TLS handshake protocol allows authentication of one or both peers using static, long-term credentials.
In some cases, it is also desirable to ensure that the peer runtime environment is in a secure state.

> Attestation is the process by which an entity produces evidence about itself that another party can use to appraise whether that entity is found in a secure state.

Author: ```suggestion
Such an assurance can be achieved using attestation which is a process by which an entity produces evidence about itself that another party can use to appraise whether that entity is found in a secure state.
```

This document describes a series of protocol extensions to the TLS 1.3 handshake that enables the binding of the TLS authentication key to a remote attestation session.
This enables an entity capable of producing attestation evidence, such as a confidential workload running in a Trusted Execution Environment (TEE), or an IoT device that is trying to authenticate itself to a network access point, to present a more comprehensive set of security metrics to its peer.
These extensions have been designed to allow the peers to use any attestation technology, in any remote attestation topology, and mutually.

--- middle

#  Introduction

Attestation {{-rats-arch}} is the process by which an entity produces evidence about itself that another party can use to evaluate the trustworthiness of that entity.
This document describes a series of protocol extensions to the TLS 1.3 handshake that enables the binding of the TLS authentication key to a remote attestation session.
As a result, a peer can use "attestation credentials", consisting of compound platform evidence and key attestation, to authenticate itself to its peer during the setup of the TLS channel.

> This enables an attester, such as a confidential workload running in a Trusted Execution Environment (TEE), or an IoT device that is trying to authenticate itself to a network access point, to present a more comprehensive set of security metrics to its peer.

Author: ```suggestion
This enables an attester, such as a confidential workload running in a Trusted Execution Environment (TEE) {{-teep-arch}}, or an IoT device that is trying to authenticate itself to a network access point, to present a more comprehensive set of security metrics to its peer.
```

TLS handshake.

TIK-C, TIK-S:
: The TIK that identifies the client or the server, respectively.

"Remote attestation credentials", or
"attestation credentials", is used to refer to both attestation evidence and
attestation results, when no distinction needs to be made between them.

{::boilerplate bcp14-tagged}

# Overview

The basic functional goal is to link the authenticated key exchange of TLS with an interleaved remote attestation session in such a way that the key used to sign the handshake can be proven to be residing within the boundaries of an attested TEE.
The requirement is that the attester can provide evidence containing the security status of both the signing key and the platform that is hosting it.
The associated security goal is to obtain such binding so that no replay, relay or splicing from an adversary is possible.

> Throughout the document, we will assume the conceptual attester model described in {{Section 3 of -rats-kat}}, where TEE attestation is provided by a Platform Attestation Token (PAT) signed by the attester's "attesting environment".

Reviewer: Same comment about KAT.
